### ISSUE TITLE
Coordinate rounding changed between v5.0.0 and v5.1.0

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#2398]
For this route

`http://router.project-osrm.org/route/v1/driving/-6.808948516845703,62.0000383831454;-6.802253723144531,62.00414819760298.json`

The coordinates of the geometry changed between v5.0.0 and v5.1.0. We need to figure out if that is just a subtle different rounding or an unnecessary conversion somewhere.

**Comment by oxidase** [comment]
This is an effect of 24a75d37fbcc6c8bfc6c69b8d1178eb00c4b213e where i added rounding of normalized_lat to COORDINATE_PRECISION at https://github.com/Project-OSRM/osrm-backend/blob/master/include/util/web_mercator.hpp#L38
The results for the example are

|normalized_lat| old result         | rounded result |
|--------------|--------------|---------|
|152.000075537 |62.0000755374 |62.000076| 
|152.004142732 |62.004142732  |62.004143| 

Motivation for the change was a failing test features/testbot/compression.feature
where the distance was different by 0.2, that i thought due to truncation
```
      Tables were not identical:
          from |     to |     route |     distance |     speed
      | (-) b | (-) e | (-) abcdef,abcdef | (-) 589m | (-) 36 km/h | 
      | (+) b | (+) e | (+) abcdef,abcdef | (+) 588.8m | (+) 36 km/h | 
      | (-) e | (-) b | (-) abcdef,abcdef | (-) 589m | (-) 36 km/h | 
      | (+) e | (+) b | (+) abcdef,abcdef | (+) 588.8m | (+) 36 km/h |
```

I will try to find out how to revert only one line without reverting the complete commit.

**Comment by danpat** [comment]
@oxidase Hmm.  We don't think it's your change, because it's a) affecting longitude values as well, and b) is affecting all coordinates along the route geometry, not just the snapped PhantomNode coordinates.

A short extract from an API response:
```
-                        -6.804464,
 -                        62.000737
 +                        -6.80446,
 +                        62.00074
                      ],
                      [
 -                        -6.802269,
 -                        62.001211
 +                        -6.80227,
 +                        62.00121
                      ],
                      [
 -                        -6.800757,
 -                        62.001548
 +                        -6.80076,
 +                        62.00155
                      ],
                      [
 -                        -6.799599,
 -                        62.001805
 +                        -6.7996,
 +                        62.00181
                      ],
                      [
 -                        -6.800491,
 -                        62.002325
 +                        -6.80049,
 +                        62.00233
                      ],
```

**Comment by oxidase** [comment]
@danpat, you are right, looks like my change  affects only least significant digit of latitude.
But i see from your diff  that coordinate precision is changed from by one order of magnitude.
Please could you give the query and bad/good sha1?

**Comment by danpat** [comment]
@oxidase It's somewhere between the 5.0.2 and 5.1.0 tags.  We filed this ticket when we noticed the problem, but we haven't had time to really try to figure out what's going on yet.

**Comment by TheMarex** [comment]
This was fixed with #2408.
